src/core/SampleBuffer.cpp

```diff
@@ -271,31 +279,40 @@ void SampleBuffer::update(bool keepSettings)
 		m_frames = 0;
 
 		const QFileInfo fileInfo(file);
-		if (fileInfo.size() > fileSizeMax * 1024 * 1024)
+		if (!fileInfo.isReadable())
 		{
-			fileLoadError = true;
+			fileLoadError = ReadPermissionDenied;
+		}
+		else if (fileInfo.size() > fileSizeMax * 1024 * 1024)
+		{
+			fileLoadError = TooLarge;
 		}
 		else
 		{
 			// Use QFile to handle unicode file names on Windows
 			QFile f(file);
-			SNDFILE * sndFile;
+			SNDFILE * sndFile = nullptr;
 			SF_INFO sfInfo;
 			sfInfo.format = 0;
+
 			if (f.open(QIODevice::ReadOnly) && (sndFile = sf_open_fd(f.handle(), SFM_READ, &sfInfo, false)))
 			{
 				f_cnt_t frames = sfInfo.frames;
 				int rate = sfInfo.samplerate;
 				if (frames / rate > sampleLengthMax * 60)
 				{
-					fileLoadError = true;
+					fileLoadError = TooLarge;
 				}
 				sf_close(sndFile);
 			}
+			else if (sndFile == nullptr)
```

Comment:
This condition will always evaluate as true - the only way `sndFile` can be assigned something other than `nullptr` is in the first condition, but if that happens, then that condition will be true and the first block will run instead.